test/shared/src/main/scala/zio/test/DefaultTestReporter.scala

```diff
@@ -139,6 +139,7 @@ object DefaultTestReporter {
 
   private def renderCause(cause: Cause[Any], offset: Int): String =
     cause match {
+      case Cause.Die(AssertionFailure(message))     => message
```

Comment:
@adamgfraser @ioleo 

What about consistently using `null` stack trace to indicate special rendering (i.e. rendering just the message)? Then maybe we can reduce never-ending lists of special cases?

Reply:
@jdegoes Yes I think that is definitely the direction we want to go in. See above comments from @neko-kai that I think get us to the same place with `Cause`.